### PR TITLE
search: separate setting query and repo cache invalidation

### PR DESCRIPTION
**Pull request by rvantonder**
Semantics preserving and setup for a larger change. Previously I used a `setQuery` method to set a subexpression and invalidate repo cache as needed. This was to ensure that we performed these two actions together. Now, I will be breaking the need to set the query state in the resolver or track its state there, which has always been thorny. To do that, I need to first pull out the part that depends on the resolver method.

### CODE REVIEW

File: cmd/frontend/graphqlbackend/search_results.go [hunk 1]
```diff
@@ -707,11 +705,13 @@ func (r *searchResolver) evaluatePatternExpression(ctx context.Context, q query.
 		case query.Or:
 			return r.evaluateOr(ctx, q)
 		case query.Concat:
-			r.setQuery(q.ToParseTree())
+			r.invalidateCache()
+			r.Query = q.ToParseTree()
```
Comment (rvantonder) [inline cmd/frontend/graphqlbackend/search_results.go:709]: This line and others like it will disappear in changes over the next couple days.

**Comment by rvantonder** [comment]
Adding @tsenart as FYI on these changes, let me know if you don't want that :-)

**Review by keegancsmith — APPROVED** [review]

**Comment by tsenart** [comment]
Glad to review these!!

**Review by tsenart — APPROVED** [review]

**Pull request merged**